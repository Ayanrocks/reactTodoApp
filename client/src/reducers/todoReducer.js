import { FETCH_TODOS, FETCH_BUCKETS } from "../actions/types";

const initialState = {
  todos: [],
  buckets: []
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TODOS:
      return { ...state, todos: action.payload };
    case FETCH_BUCKETS:
      return { ...state, buckets: action.payload };
    default:
      return state;
  }
};
