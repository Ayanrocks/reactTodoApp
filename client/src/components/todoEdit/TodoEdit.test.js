import React from "react";
import TodoEdit from "./index";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "../../reducers";

it("TodoEdit renders correctly", () => {
  const mockStore = createStore(reducers, {}, applyMiddleware(reduxThunk));
  const tree = renderer
    .create(
      <Provider store={mockStore}>
        <TodoEdit />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
