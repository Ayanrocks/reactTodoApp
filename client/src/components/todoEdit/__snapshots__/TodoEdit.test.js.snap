// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoEdit renders correctly 1`] = `
<div
  className="my-5"
>
  <div
    className="container my-5"
  >
    <div
      className="row"
    >
      <div
        className="form form-card"
      >
        <div
          className="col-md"
        >
          <button
            className="btn btn-primary btn-custom"
            onClick={[Function]}
          >
            <span
              className="fa fa-pencil"
            />
          </button>
          <button
            className="btn btn-danger btn-custom"
            onClick={[Function]}
          >
            <span
              className="fa fa-trash"
            />
          </button>
        </div>
        <div
          className="col-md"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="name"
            >
              Todo
            </label>
            <p>
              
            </p>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="status"
            >
              Status
            </label>
            <p>
              
            </p>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="Bucket"
            >
              Bucket
            </label>
            <p>
              
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
